torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1785,9 +1939,24 @@ def __init__(
         self.frozen = False
         self.dim_constraints: Optional[DimConstraints] = None
 
+        if _translation_validator_enabled():
+            self.validator = TranslationValidator()
+
     def freeze(self):
         self.frozen = True
 
+    def _add_input_guard(self, expr: sympy.Expr) -> None:
+        if _translation_validator_enabled():
+            self.validator.add_input(expr)
+
+    def _add_output_guard(self, expr: sympy.Expr) -> None:
+        if _translation_validator_enabled():
+            self.validator.add_output(expr)
+
+    def _check_translation_validate(self) -> None:
+        if _translation_validator_enabled():
+            assert self.validator.validate(), "translation validation failed."
```

Comment:
it might be better to just throw an exception with all the info it, rather than breadcrumbing it with logs. Not difficult to do.